core/src/Kokkos_Macros.hpp

```diff
@@ -652,4 +652,13 @@
 #define KOKKOS_THREAD_LOCAL __thread
 #endif
 
+#define KOKKOS_PROFILING_STRINGIFY(x) #x
+#define KOKKOS_PROFILING_TOSTRING(x) KOKKOS_PROFILING_STRINGIFY(x)
+#if defined(KOKKOS_PROFILING_ENABLE_SOURCE_INFO)
+#define KOKKOS_PROFILING_NAME(name)              \
+  (__FILE__ "KKSPLIT" KOKKOS_PROFILING_TOSTRING( \
+      __LINE__) "KKSPLIT" KOKKOS_PROFILING_TOSTRING(name))
+#else
+#define KOKKOS_PROFILING_NAME(name) (KOKKOS_PROFILING_TOSTRING(name))
```

Comment:
How about KOKKOS_FILELINE_NAME or something like this? PROFILING is probably wrong. Also we should think about the namespacing with Profiling in general since we also use this for debugging and some of it potentially as part of tuning too.

Reply:
Done